tests/infer/test_valid_models.py

```diff
@@ -1613,16 +1613,22 @@ def guide():
 
 def test_tail_adaptive():
 
-    def model():
+    def plateless_model():
         pyro.sample("x", dist.Normal(0., 1.))
 
+    def plate_model():
+        x = pyro.sample("x", dist.Normal(0., 1.))
+        with pyro.plate('observe_data'):
+            pyro.sample('obs', dist.Normal(x, 1.0), obs=torch.arange(5).type_as(x))
+
     def rep_guide():
         pyro.sample("x", dist.Normal(0., 2.))
 
     def nonrep_guide():
         pyro.sample("x", fakes.NonreparameterizedNormal(0., 2.))
 
-    assert_ok(model, rep_guide, TraceTailAdaptive_ELBO(vectorize_particles=True, num_particles=2))
-    assert_error(model, rep_guide, TraceTailAdaptive_ELBO(vectorize_particles=False, num_particles=2))
-    assert_warning(model, rep_guide, TraceTailAdaptive_ELBO(vectorize_particles=True, num_particles=1))
-    assert_error(model, nonrep_guide, TraceTailAdaptive_ELBO(vectorize_particles=True, num_particles=2))
+    assert_ok(plateless_model, rep_guide, TraceTailAdaptive_ELBO(vectorize_particles=True, num_particles=2))
```

Comment:
FYI it's generally useful to split assertions into separate tests so that a failure of the first does not prevent testing of all assertions following. Often we can do this with parametrized tests. Feel free to split this up or not.